apps/server/src/modules/files-storage/service/file-storage.service.spec.ts

```diff
@@ -0,0 +1,188 @@
+import { createMock, DeepMocked } from '@golevelup/ts-jest';
+import { MikroORM } from '@mikro-orm/core';
+import { ObjectId } from '@mikro-orm/mongodb';
+import { Test, TestingModule } from '@nestjs/testing';
+import { EntityId, FileRecord, FileRecordParentType } from '@shared/domain';
+import { FileRecordRepo } from '@shared/repo';
+import { fileRecordFactory, setupEntities } from '@shared/testing';
+import { Logger } from '@src/core/logger';
+import { S3ClientAdapter } from '../client/s3-client.adapter';
+import { FileRecordParams } from '../controller/dto/file-storage.params';
+import { FilesStorageHelper } from '../helper';
+import { FilesStorageService } from './file-storage.service';
+
+describe('FilesStorageService', () => {
+	let module: TestingModule;
+	let service: FilesStorageService;
+	let fileRecordRepo: DeepMocked<FileRecordRepo>;
+	let storageClient: DeepMocked<S3ClientAdapter>;
+	let filesStorageHelper: DeepMocked<FilesStorageHelper>;
+	let orm: MikroORM;
+	let fileRecord: FileRecord;
+	let fileRecords: FileRecord[];
+
+	const entityId: EntityId = new ObjectId().toHexString();
+	const userId: EntityId = new ObjectId().toHexString();
+	const schoolId: EntityId = new ObjectId().toHexString();
+
+	beforeAll(async () => {
+		orm = await setupEntities();
+
+		fileRecord = fileRecordFactory.buildWithId({ name: 'text.txt' });
+	});
+
+	afterAll(async () => {
+		await orm.close();
+	});
+
+	beforeEach(async () => {
+		module = await Test.createTestingModule({
+			providers: [
+				FilesStorageService,
+				{
+					provide: S3ClientAdapter,
+					useValue: createMock<S3ClientAdapter>(),
+				},
+				{
+					provide: FileRecordRepo,
+					useValue: createMock<FileRecordRepo>(),
+				},
+				{
+					provide: Logger,
+					useValue: createMock<Logger>(),
+				},
+				{
+					provide: FilesStorageHelper,
+					useValue: createMock<FilesStorageHelper>(),
+				},
+			],
+		}).compile();
+
+		service = module.get(FilesStorageService);
+		storageClient = module.get(S3ClientAdapter);
+		fileRecordRepo = module.get(FileRecordRepo);
+		filesStorageHelper = module.get(FilesStorageHelper);
+
+		fileRecords = [
+			fileRecordFactory.buildWithId({ parentId: userId, schoolId, name: 'text.txt' }),
+			fileRecordFactory.buildWithId({ parentId: userId, schoolId, name: 'text-two.txt' }),
+			fileRecordFactory.buildWithId({ parentId: userId, schoolId, name: 'text-tree.txt' }),
+		];
+		fileRecordRepo.findBySchoolIdAndParentId.mockResolvedValue([fileRecords, fileRecords.length]);
+
+		fileRecordRepo.save.mockImplementation((entity: FileRecord | FileRecord[]) => {
+			(entity as FileRecord).id = entityId;
+			return Promise.resolve();
+		});
+	});
+
+	afterEach(async () => {
+		await module.close();
+	});
+
+	it('should be defined', () => {
+		expect(service).toBeDefined();
+	});
+
+	describe('deleteFilesOfParent()', () => {
+		let requestParams: FileRecordParams;
```

Comment:
Let me check the latest decision. But i think we have commited us to use setup methodes with return for modifing env.
If the value never changed it it fine to use beforeEach. If the value is changed we put a 
`
const setup = () => { ....return FileRecordParams } on this place. It can be called inside of the it()`s.
`